aiida_testing/export_cache/_fixtures.py

```diff
@@ -0,0 +1,375 @@
+    def _load_cache(path_to_cache=None, node=None, load_all=False):
+        """
+        Function to import an AiiDA graph
+
+        :param path_to_cache: str or path to the AiiDA export file to load,
+            if path_to_cache points to a directory, all import files in this dir are imported
+
+        :param node: AiiDA node which cache to load,
+            if no path_to_cache is given tries to guess it.
+        """
+        from aiida.tools.importexport import import_data
+
+        if path_to_cache is None:
+            if node is None:
+                raise ValueError(
+                    "Node argument can not be None "
+                    "if no explicit 'path_to_cache' is specified"
+                )
+            #else:  # create path from node
+            #    pass
+            #    # get default data dir
+            #    # get hash for give node
+            #    # construct path from that
+        else:
+            # TODO: what about relative path to data_dir
+            full_import_path = pathlib.Path(path_to_cache)
+
+        if full_import_path.exists():
+            if os.path.isfile(full_import_path):
+                # import cache, also import extras
+                import_data(full_import_path, extras_mode_existing='ncu', extras_mode_new='import')
+            elif os.path.isdir(full_import_path):
+                for filename in os.listdir(full_import_path):
+                    file_full_import_path = os.path.join(full_import_path, filename)
+                    # we curretly assume all files are valid aiida exports...
+                    # maybe check if valid aiida export, or catch exception
+                    import_data(
+                        file_full_import_path, extras_mode_existing='ncu', extras_mode_new='import'
+                    )
+            else:  # Should never get there
+                raise OSError(
+                    "Path: {} to be imported exists, but is neither a file or directory.".
+                    format(full_import_path)
+                )
+        else:
+            raise OSError("File: {} to be imported does not exist.".format(full_import_path))
+
+        # need to rehash after import, otherwise cashing does not work
+        # for this we rehash all process nodes
+        # this way we use the full caching mechanism of aiida-core.
+        # currently this should only cache CalcJobNodes
+        qub = QueryBuilder()
+        qub.append(ProcessNode)  # query for all ProcesNodes
+        to_hash = qub.all()
+        for node1 in to_hash:
+            node1[0].rehash()
+
+    return _load_cache
+
+
+@pytest.fixture(scope='function')
+def with_export_cache(export_cache, load_cache):
+    """
+    Fixture to use in a with() environment within a test to enable caching within the with.
+    Requires to provide an absolutpath to the export file to load or export to.
+    """
+    @contextmanager
+    def _with_export_cache(data_dir_abspath, calculation_class=None):
```

Comment:
if we have a default value for the data dir in `run_with_cache` (line 299) should we have the same here?

Reply:
I'd actually prefer not having a default in either of the two places.

Reply:
I don't have strong feelings towards one or the other. It should just be the same for both ways